api/src/main/java/com/javadiscord/jdi/internal/api/DiscordRequestDispatcher.java

```diff
@@ -21,6 +22,7 @@ public class DiscordRequestDispatcher implements Runnable {
     private static final Logger LOGGER = LogManager.getLogger();
     private final BlockingQueue<DiscordRequestBuilder> queue;
     private final String botToken;
+    private AtomicBoolean running;
```

Comment:
It shouldnt be `null` by default, instead it should be `new AtomicBoolean(false)` and then in the run just use the set method at the beginning